### DISCUSSION TITLE
Can I ask how node-redid library works and one example

### DISCUSSION THREAD

**Discussion by kaushal9808** [NodeRedis/node_redis#2475] [Q&A]
Hello Team,

I am looking for improvement, for command like Zrank and ZrevRank so every one can get benefit,

Can anyone please tell us how it library work its internally, Zrank.d.ts how it will integrate further,

if there is any help that will be great. 

need to know internal working first then we will enhance the command feature.

**Comment by leibale** [comment]
Hi, sorry for the huge delay...

Looks like [ZRANK](https://github.com/redis/node-redis/blob/master/packages/client/lib/commands/ZRANK.ts) and [ZREVRANK](https://github.com/redis/node-redis/blob/master/packages/client/lib/commands/ZREVRANK.ts) are already implemented...

**Reply by kaushal9808** [reply]
Hello Leibale, ZREVRANK only provide the rank information but we are here to enhance the functionality of the ZREVRANK so every one can relate real life experience in gaming area. like we can implement zrevrank with enhance functionality in one round trip with member and score information with rank. so every one use that function with real based scenario. and improve performance in one round trip with full information which is required by lots of real application.

**Reply by leibale** [reply]
Oh, you want to add support for the `WITHSCORE` option? [There is already an open pull request for that](https://github.com/redis/node-redis/pull/2489), and I'll merge it as soon as Redis 7.2 will be released.

**Reply by kaushal9808** [reply]
ok thanks, when redis is planning to release its latest version any update ?

**Reply by leibale** [reply]
I'm not too sure.. @itamarhaber might know..

**Reply by itamarhaber** [reply]
The release is planned for June 2023.
